### PR TITLE
Updated requests for the UI

### PR DESCRIPTION
**Pull request by harmochiky**
Summary 

Updated search requests for UI in the front end.

Removes the old search and customSearch endpoints

Dependent on the branch https://github.com/jembi/JeMPI/tree/CU-86by86ygn_Refactor-Search-endpoints
https://app.clickup.com/t/86by86ygn
https://github.com/jembi/JeMPI/pull/234

### CODE REVIEW

**Comment by rcrichton** [comment]
Task linked: [CU-86by86ygn Refactor Search endpoints](https://app.clickup.com/t/86by86ygn)
Task linked: [CU-86by87zjf Update requests for UI](https://app.clickup.com/t/86by87zjf)

File: JeMPI_Apps/JeMPI_UI/src/services/apiRoutes.ts [hunk 1]
```diff
@@ -11,13 +11,18 @@ const ROUTES = {
   POST_UPDATE_NOTIFICATION: '/notificationRequest',
   POST_SIMPLE_SEARCH: '/search',
   POST_CUSTOM_SEARCH: '/customSearch',
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_UI/src/services/apiRoutes.ts:13]: @harmochiky please remove these old endpoints and update the mockbackend dependency

File: JeMPI_Apps/JeMPI_UI/src/services/ApiClient.ts [hunk 2]
```diff
@@ -214,8 +214,15 @@ export class ApiClient {
...
-    }/${isGoldenOnly ? 'golden' : 'patient'}`
+      isGoldenOnly
+        ? isCustomSearch
+          ? ROUTES.POST_CUSTOM_GOLDEN_SEARCH
+          : ROUTES.POST_SIMPLE_GOLDEN_SEARCH
+        : isCustomSearch
+        ? ROUTES.POST_SIMPLE_INTERACTION_PATIENT_SEARCH
+        : ROUTES.POST_CUSTOM_INTERACTION_PATIENT_SEARCH
+    }`
+
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_UI/src/services/ApiClient.ts:225]: @harmochiky please review the use of the interaction patient search. To my knowledge, the ui does not use this anymore. If so, please remove this and the need for checking on isGoldenOnly property

**Review by MatthewErispe — CHANGES_REQUESTED** [review]

File: JeMPI_Apps/JeMPI_UI/src/services/ApiClient.ts (hunk 2 above)
Comment (harmochiky) [reply JeMPI_Apps/JeMPI_UI/src/services/ApiClient.ts]: Resolved @MatthewErispe

**Review by MatthewErispe — APPROVED** [review]
@harmochiky I am happy with your changes. I also made a few additional changes

**Pull request merged**